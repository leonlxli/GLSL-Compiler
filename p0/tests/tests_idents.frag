validstring
_valid
valid_string123
weird*string

EXACTLY31charsshouldNOTtruncate

EXACTLY32charsshouldtruncatessss

dasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadkjhfjkdsahfjksadhfjkdshfjksdahfkjahjfksdahfkjahsfdasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadkjhfjkdsahfjksadhfjkdshfjksdahfkjahjfksdahfkjahsfdasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadkjhfjkdsahfjksadhfjkdshfjksdahfkjahjfksdahfkjahsfdasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadkjhfjkdsahfjksadhfjkdshfjksdahfkjahjfksdahfkjahsfdasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadkjhfjkdsahfjksadhfjkdshfjksdahfkjahjfksdahfkjahsfdasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadkjhfjkdsahfjksadhfjkdshfjksdahfkjahjfksdahfkjahsfdasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadkjhfjkdsahfjksadhfjkdshfjksdahfkjahjfksdahfkjahsfdasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadkjhfjkdsahfjksadhfjkdshfjksdahfkjahjfksdahfkjahsfdasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadk

thisshouldNOTreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthi

thisSHOULDreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreport

true