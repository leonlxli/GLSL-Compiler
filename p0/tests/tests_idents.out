validstring  line 1 cols 1-11 is T_Identifier 
_valid       line 2 cols 1-6 is T_Identifier 
valid_string123 line 3 cols 1-15 is T_Identifier 
weird        line 4 cols 1-5 is T_Identifier 
*            line 4 cols 6-6 is '*' 
string       line 4 cols 7-12 is T_Identifier 
EXACTLY31charsshouldNOTtruncate line 6 cols 1-31 is T_Identifier 
EXACTLY32charsshouldtruncatessss line 8 cols 1-32 is T_Identifier (truncated to EXACTLY32charsshouldtruncatesss)

*** Error line 10.
*** Identifier too long: "dasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadkjhfjkdsahfjksadhfjkdshfjksdahfkjahjfksdahfkjahsfdasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadkjhfjkdsahfjksadhfjkdshfjksdahfkjahjfksdahfkjahsfdasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadkjhfjkdsahfjksadhfjkdshfjksdahfkjahjfksdahfkjahsfdasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadkjhfjkdsahfjksadhfjkdshfjksdahfkjahjfksdahfkjahsfdasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadkjhfjkdsahfjksadhfjkdshfjksdahfkjahjfksdahfkjahsfdasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadkjhfjkdsahfjksadhfjkdshfjksdahfkjahjfksdahfkjahsfdasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadkjhfjkdsahfjksadhfjkdshfjksdahfkjahjfksdahfkjahsfdasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadkjhfjkdsahfjksadhfjkdshfjksdahfkjahjfksdahfkjahsfdasklfjlkajflkdsjalfkjadskljflkadsjflkasdjfkljdksahfkjhadsjkfhjkdshfkjshdafkjhasdjkfhsadk"

thisshouldNOTreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthisshouldnotreportanerroritisEXACTLY1023CHARACTERSthi line 12 cols 1-1023 is T_Identifier (truncated to thisshouldNOTreportanerroritisE)

*** Error line 14.
*** Identifier too long: "thisSHOULDreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreportanerrorITISEXACTLY1024CHARACTERSthisshouldreport"

true         line 16 cols 1-4 is T_BoolConstant (value = true)
